### PR TITLE
CLOUDP-263653: Add Spectral validation for Postman

### PR DESCRIPTION
**Pull request by Luke-Sanderson**
## Proposed changes

- Add Spectral rules modified from [Postmanlabs spectral linter](https://github.com/postmanlabs/postman-collection-spectral-linter/tree/main)
- Run Spectral in Optional Validation

Tested changes on Fork

_Jira ticket:_ CLOUDP-263653

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tools/postman/validation/spectral.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      field: description
+      function: truthy
+
+  path-parameters-require-description:
+    description: All request path parameters must have a description.
+    given: $..request.url.variable[*]
+    severity: error
+    then:
+      field: description
+      function: truthy
```
Comment (gssbzn) [inline tools/postman/validation/spectral.yaml:81]: how are these different from https://github.com/mongodb/openapi/blob/0819d1d3d007c4498e10db1518b977b873bdd28b/.spectral.yaml#L41-L56

File: tools/postman/validation/functions/patchRequestShouldHaveBody.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export default (request) => {
+  if (!request.method || request.method !== "PATCH") {
```
Comment (jeroenvervaeke) [inline tools/postman/validation/functions/patchRequestShouldHaveBody.js:2]: Isn't this the same as
```js
if (request.method !== "PATCH") {
```

File: tools/postman/validation/spectral.yaml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+  http-should-not-be-used:
+    description: HTTP should not be used.
+    message: "HTTP should not be used: {{value}}"
+    given: $..url.raw
+    severity: warn
+    then:
+      function: pattern
+      functionOptions:
+        notMatch: "/^http:/i"
```
Comment (gssbzn) [inline tools/postman/validation/spectral.yaml:57]: I believe this comes out of the box if you include the default rules https://github.com/mongodb/openapi/blob/0819d1d3d007c4498e10db1518b977b873bdd28b/.spectral.yaml#L2-L3

File: tools/postman/validation/functions/patchRequestShouldHaveBody.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    return;
+  }
+  const body = request.body;
+  if (!body || !body.raw) {
```
Comment (jeroenvervaeke) [inline tools/postman/validation/functions/patchRequestShouldHaveBody.js:6]: not sure if you can use `?.` in the version of js you're using, but if you can you can replace it by
```js
if (!body?.raw) {
```

File: tools/postman/validation/spectral.yaml (hunk 1 above)
Comment (Luke-Sanderson) [reply tools/postman/validation/spectral.yaml:81]: There is no difference, I will remove them

File: tools/postman/validation/spectral.yaml (hunk 3 above)
Comment (Luke-Sanderson) [reply tools/postman/validation/spectral.yaml:57]: I didn't see a rule in here that corresponds so I will keep it for now
https://docs.stoplight.io/docs/spectral/4dec24461f3af-open-api-rules#openapi-rules

File: tools/postman/validation/functions/patchRequestShouldHaveBody.js (hunk 4 above)
Comment (Luke-Sanderson) [reply tools/postman/validation/functions/patchRequestShouldHaveBody.js]: Yes you can, thank you

File: tools/postman/validation/functions/patchRequestShouldHaveBody.js (hunk 2 above)
Comment (Luke-Sanderson) [reply tools/postman/validation/functions/patchRequestShouldHaveBody.js]: Added

**Pull request merged**